### DISCUSSION TITLE
DynUNet output dimension

### DISCUSSION THREAD

**Discussion by LSnyd** [Project-MONAI/tutorials#543] [General]
Hi all, 

I build a 3D Segmentation pipeline that does not perform well, but at least runs with UNet. 
I now want to switch to DynUNet, but the output dimensions are different. Even though I configure them with the same output dimension. 

For UNet my output dimensions are [4, 2, 96, 96, 96], while for DynUNet its [4, 4, 2, 96, 96, 96].

The networks are configured the following way: 

```
model = DynUNet(
        spatial_dims=3,
        in_channels=1,
        out_channels=2,
        kernel_size=kernels,
        strides=strides,
        upsample_kernel_size=strides[1:],
        norm_name="batch",
        deep_supervision=True,
        deep_supr_num=deep_supr_num[task_id],
    ).to(device)

model = monai.networks.nets.UNet(
        spatial_dims=3,
        in_channels=1,
        out_channels=2,
        channels=(16, 32, 64, 128, 256),
        strides=(2, 2, 2, 2),
        num_res_units=2,
).to(device)
```

Does anybody know where the extra dimension in the output of DynUnet comes from?

**Comment by Nic-Ma** [comment]
Hi @LSnyd ,

Thanks for your interest here.
The reason is because you set `deep_supervision=True`, so it also outputs the feature maps:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/nets/dynunet.py#L267

Thanks.

**Reply by LSnyd** [reply]
Hi @Nic-Ma, 

thanks a lot for the fast reply, that quickly solved my problem! 

Thanks!
